pymatgen/io/shengbte.py

```diff
@@ -0,0 +1,156 @@
+                of 1.0 is theoretically guaranteed to work, but significant
+                speedups can sometimes be achieved by reducing it with
+                negligible loss of precision.
+            t (int or float): temperature (Kelvin)
+            **kwargs: Other ShengBTE parameters. Several parameters are required
+                for ShengBTE to run - we have listed these parameters below:
+                - nelements (int): number of different elements in the compound
+                - natoms (int): number of atoms in the unit cell
+                - lattvec (size 3x3 array): real-space lattice vectors, in units
+                  of lfactor
+                - types (size natom list): a vector of natom integers, ranging
+                  from 1 to nelements, assigning an element to each atom in the
+                  system
+                - elements (size natom list): a vector of element names
+                - positions (size natomx3 array): atomic positions in lattice
+                  coordinates
+                - scell (size 3 list): supercell sizes along each crystal axis
+                  used for the 2nd-order force constant calculation
+        """
+        if ngrid is None:
+            ngrid = [25, 25, 25]
+
+        self["ngrid"] = ngrid
+        self["lfactor"] = lfactor
+        self["scalebroad"] = scalebroad
+        self["t"] = t
+        self.update(kwargs)
+
+    @classmethod
+    def from_file(cls, filepath):
+        """
+        Read a CONTROL namelist file and output a 'Control' object
+
+        Args:
+            filepath (String): Path of the CONTROL file.
+
+        Returns:
+            'Control' object with parameters instantiated.
+        """
+        nml = f90nml.read(filepath)
+        sdict = nml.todict()
+
+        all_dict = {}
+        all_dict.update(sdict["allocations"])
+        all_dict.update(sdict["crystal"])
+        all_dict.update(sdict["parameters"])
+        all_dict.update(sdict["flags"])
+
+        return cls.from_dict(all_dict)
+
+    @classmethod
+    def from_dict(cls, sdict):
+        """
+        Write a CONTROL file from a Python dictionary.
+        Description and default parameters can be found at
+        https://bitbucket.org/sousaw/shengbte/src/master/.
+        Note some parameters are mandatory. Optional parameters
+        default here to None and will not be written to file.
+
+        Args:
+            dict: A Python dictionary of ShengBTE input parameters.
+        """
+        return cls(**sdict)
+
+    def to_file(self, filename):
+        """
+        Writes ShengBTE CONTROL file from 'Control' object
+        """
+
+        for param in self.required_params:
+            if param not in self.as_dict():
+                warnings.warn(
+                    "Required parameter '{}' not specified!".format(param))
+                raise AttributeError("param {} not specified".format(param))
```

Comment:
Remove this line. We already give a warning. If a user wants to make an invalid ShengBTE input file they should be able to.